pkg/cli/klothomain.go

```diff
@@ -0,0 +1,315 @@
+	"os"
+
+	"github.com/klothoplatform/klotho/pkg/updater"
+
+	"github.com/klothoplatform/klotho/pkg/input"
+
+	"github.com/fatih/color"
+	"github.com/klothoplatform/klotho/pkg/analytics"
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/logging"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type KlothoMain struct {
+	UpdateStream     string
+	Version          string
+	VersionQualifier string
+	PluginSetup      func(*PluginSetBuilder) error
+}
+
+var klothoUpdater = updater.Updater{ServerURL: updater.DefaultServer, Stream: updater.DefaultStream}
+
+var cfg struct {
+	verbose       bool
+	config        string
+	outDir        string
+	ast           bool
+	caps          bool
+	provider      string
+	appName       string
+	strict        bool
+	disableLogo   bool
+	internalDebug bool
+	version       bool
+	uploadSource  bool
+	update        bool
+	cfgFormat     string
+	login         string
+}
+
+var hadWarnings = atomic.NewBool(false)
+var hadErrors = atomic.NewBool(false)
+
+func init() {
+	err := zap.RegisterEncoder("klotho-cli", func(zcfg zapcore.EncoderConfig) (zapcore.Encoder, error) {
+		return logging.NewConsoleEncoder(cfg.verbose, hadWarnings, hadErrors), nil
+	})
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+const (
+	defaultOutDir = "compiled"
+)
+
+func (km KlothoMain) Main() {
+
+	var root = &cobra.Command{
+		Use:  "klotho [path to source]",
+		RunE: km.run,
+	}
+
+	flags := root.Flags()
+
+	flags.BoolVarP(&cfg.verbose, "verbose", "v", false, "Verbose flag")
+	flags.StringVarP(&cfg.config, "config", "c", "", "Config file")
+	flags.StringVarP(&cfg.outDir, "outDir", "o", defaultOutDir, "Output directory")
+	flags.BoolVar(&cfg.ast, "ast", false, "Print the AST to a companion file")
+	flags.BoolVar(&cfg.caps, "caps", false, "Print the capabilities to a companion file")
+	flags.StringVarP(&cfg.cfgFormat, "cfg-format", "F", "yaml", "The format for the compiled config file (if --config is not specified). Supports: yaml, toml, json")
+	flags.StringVar(&cfg.appName, "app", "", "Application name")
+	flags.StringVarP(&cfg.provider, "provider", "p", "", fmt.Sprintf("Provider to compile to. Supported: %v", "aws"))
+	flags.BoolVar(&cfg.strict, "strict", false, "Fail the compilation on warnings")
+	flags.BoolVar(&cfg.disableLogo, "disable-logo", false, "Disable printing the Klotho logo")
+	flags.BoolVar(&cfg.uploadSource, "upload-source", false, "Upload the compressed source folder for debugging")
+	flags.BoolVar(&cfg.internalDebug, "internalDebug", false, "Enable debugging for compiler")
+	flags.BoolVar(&cfg.version, "version", false, "Print the version")
+	flags.BoolVar(&cfg.update, "update", false, "update the cli to the latest version")
+	flags.StringVar(&cfg.login, "login", "", "Login to Klotho with email. For anonymous login, use 'local'")
+	_ = flags.MarkHidden("internalDebug")
```

Comment:
this will no longer be hidden really, but i guess it really isnt today

Reply:
I didn't quite understand that bit, actually. In what way was it hidden before?

This code shouldn't have any functional changes, but I can verify — I'm just not sure what the user-visible thing to verify is.

Reply:
just if they do --help, but if they look at the source code they would see it no matter what